build/.eslintrc.yaml

```diff
@@ -1,3 +1,4 @@
+extends: '@wfcd/eslint-config/esm'
 rules:
   no-console: off
   import/no-extraneous-dependencies: off
```

Comment:
:thinking: was this added because of the other parts? this should be covered by our rule for extraneous deps where we allow dev dependencies to be used in non-spec code :thinking: i'm thinking we need to update the matching in the esm to match .mjs and normal js

Reply:
i think v1.3.1 fixes this from being needed

Reply:
(i swear this is the last thing.... i think)

Reply:
(and rebasing the branch)

Reply:
If we're working with esm in the build why the rules should be for javascript?

It's not because of dev dependencies which was also an issue but in the tests, this is for the rule that i changed specifically for esm, esm modules work best when specifying the file extension peer node and browser specs. That's why every local import has the extension of the file generally `.mjs`, but eslint upstream rule wouldn't allow it so i changed from:
```js
'import/extensions': 0
```
to
```js
'import/extensions': ['error', 'always']
```

[the node specifications for esm extension resolution](https://nodejs.org/docs/latest-v16.x/api/esm.html#mandatory-file-extensions)

Reply:
I also don't understand why a rebase should be needed

Reply:
> 🤔 why was this added? this should be covered by our rule for extraneous deps where we allow dev dependencies to be used in non-spec code 🤔 i'm thinking we need to update the matching in the esm to match .mjs and normal js

or maybe you mean `import/no-extraneous-dependencies: off`? that it should be removed?

Reply:
correct, should be able to remove `import/no-extraneous-dependencies: off` with the latest update. i think what you're doing in always giving the extension is fine and makes perfect sense. i was commenting specifically on the line about `import/no-extraneous-dependencies: off`

Reply:
makes sense my bad

Reply:
The rule isn't exactly right in there it should be `build/**/*.mjs` not exactly sure why. And also shouldn't we put this rules in the eslint config of warframe-items as it's specific to it? Or is there other projects that also benefit from it?

Reply:
like
```yaml
  import/no-extraneous-dependencies:
    - error
    - devDependencies:
        - "build/**/*.mjs"
```

Reply:
it's related to patchlogs as well, since they both use the same build output folder...

which part doesn't work? is it getting overridden in the esm config and i'm just silly?

Reply:
it's odd that it's an issue with it now but didn't have an issue with it before 🤔 let's still update the base config, that's why i have it there :D

Reply:
should publish here in a sec